dartagnan/src/main/java/com/dat3m/dartagnan/wmm/relation/base/stat/StaticRelation.java

```diff
@@ -47,4 +55,35 @@ protected BooleanFormula encodeApprox(SolverContext ctx) {
         }
         return enc;
     }
+
+    protected void addMatchingTupleSet(List<Event> events, String open, String close, Consumer<Tuple> outMust) {
+        ExecutionAnalysis exec = analysisContext.get(ExecutionAnalysis.class);
+        // assume order by cId
+        // assume cId describes a topological sorting over the control flow
+        for(int end = 1; end < events.size(); end++) {
+            Event store = events.get(end);
+            if(!store.is(close)) {
+                continue;
+            }
+            int start = range(end - 1, -1).filter(i -> exec.isImplied(store, events.get(i))).findFirst().orElse(0);
```

Comment:
This doesn't work and will always default to `0`. If you want to have a stream that iterates backwards, you need to use `IntStream.iterate(end - 1, i -> i >= 0, i -> i - 1)`.

Reply:
Thanks, that's correct.